spec/models/survey_phrase_spec.rb

```diff
@@ -1,5 +1,53 @@
-require 'rails_helper'
+require 'spec_helper'
 
 RSpec.describe SurveyPhrase, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'mentions by date range for phrase' do
+    context 'with empty database' do
+      it 'returns an array keyed by date and containing no mentions' do
+        start_date = Date.new(2020, 3, 10)
+        end_date = Date.new(2020, 3, 11)
+
+        result = SurveyPhrase.mentions_by_date_range_for_phrase(nil, start_date, end_date)
+
+        expected_result = [[start_date, 0], [end_date, 0]]
+        expect(result).to eq(expected_result)
+      end
+    end
+
+    context 'with populated database' do
+      before :each do
+        @phrase = FactoryBot.create(:phrase)
+
+        create_mentions_for_date('2020-03-10', 1)
+        create_mentions_for_date('2020-03-11', 3)
+        create_mentions_for_date('2020-03-13', 3)
+      end
+
+      it 'returns an array keyed by date and containing no mentions when date range matches no mentions' do
+        start_date = Date.new(2020, 3, 14)
+        end_date = Date.new(2020, 3, 15)
+
+        result = SurveyPhrase.mentions_by_date_range_for_phrase(@phrase, start_date, end_date)
+
+        expected_result = [[start_date, 0], [end_date, 0]]
+        expect(result).to eq(expected_result)
+      end
+
+      it 'returns mentions data keyed by date when data range matches mentions' do
+        start_date = Date.new(2020, 3, 10)
+        end_date = Date.new(2020, 3, 13)
+
+        result = SurveyPhrase.mentions_by_date_range_for_phrase(@phrase, start_date, end_date)
+
+        expected_result = [[start_date, 1], [Date.new(2020, 3, 11), 3], [Date.new(2020, 3, 12), 0], [end_date, 3]]
+        expect(result).to eq(expected_result)
+      end
+    end
+  end
+
+  def create_mentions_for_date(date_str, number_of_mentions)
```

Comment:
Feel free to ignore this because I'm not super familiar with factorybot and best practices but if there is a chance it might be used elsewhere I think this might want to live in the factory file. If not, I'm 50/50 about whether it should